### PR TITLE
Remove specialist topic code from specs

### PR DESCRIPTION
**Pull request by unoduetre**
## What

This prepares for https://github.com/alphagov/email-alert-api/pull/2136 and is related to https://github.com/alphagov/gds-api-adapters/pull/1250

## Why

This PR has been extracted from https://github.com/alphagov/email-alert-api/pull/2136 as requested by @hannako 

[Trello ticket](https://trello.com/c/RObMmjXG/2422-remove-specialist-topic-code-from-email-alert-api-l)

### CODE REVIEW

**Review by hannako — APPROVED** [review]

File: spec/features/daily_digest_spec.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@
         description: "Description one",
         change_note: "Change note one",
         public_updated_at: "2017-01-01 10:00:00",
-        links: { topics: [list_one_topic_id] },
+        links: { taxons: [list_one_taxon_id] },
```
Comment (hannako) [inline spec/features/daily_digest_spec.rb:39]: Just checked in the db and ContentChange uses both the key `taxons` and `taxon_tree`. No idea why! But just letting you know that it's fine to use `taxons` here.

File: spec/features/weekly_digest_spec.rb [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-  let(:list_one_topic_id) { "0eb5d0f0-d384-4f27-9da8-3f9e9b22a820" }
+  let(:list_one_taxon_id) { "0eb5d0f0-d384-4f27-9da8-3f9e9b22a820" }
   let(:list_two_taxon_id) { "6416e4e0-c0c1-457a-8337-4bf8ed9d5f80" }
 
   let(:subscriber_list_one) do
     create_subscriber_list(
       title: "Subscriber list one",
       links: {
-        topics: { any: [list_one_topic_id] },
+        taxons: { any: [list_one_taxon_id] },
```
Comment (hannako) [inline spec/features/weekly_digest_spec.rb:9]: `taxon_tree` see line 18

File: spec/integration/browsing_subscriber_lists_spec.rb [hunk 3]
```diff
@@ -11,7 +11,7 @@
         create(
           :subscriber_list,
           links: {
-            topics: { any: [uuid, "drug-device-alert"] },
+            taxons: { any: [uuid, "drug-device-alert"] },
```
Comment (hannako) [inline spec/integration/browsing_subscriber_lists_spec.rb:14]: `taxon_tree`

File: spec/features/daily_digest_spec.rb [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-  let(:list_one_topic_id) { "0eb5d0f0-d384-4f27-9da8-3f9e9b22a820" }
+  let(:list_one_taxon_id) { "0eb5d0f0-d384-4f27-9da8-3f9e9b22a820" }
   let(:list_two_taxon_id) { "6416e4e0-c0c1-457a-8337-4bf8ed9d5f80" }
 
   let(:subscriber_list_one) do
     create_subscriber_list(
       title: "Subscriber list one",
       links: {
-        topics: { any: [list_one_topic_id] },
+        taxons: { any: [list_one_taxon_id] },
```
Comment (hannako) [inline spec/features/daily_digest_spec.rb:9]: SubscriberLists use the key taxon_tree

eg
```
  id: 1447,
  title: "Education, training and skills",
  created_at: Thu, 30 Mar 2017 17:06:26.031943000 BST +01:00,
  updated_at: Wed, 20 Jan 2021 09:07:44.083612000 GMT +00:00,
  document_type: "",
  tags: {},
  links: {"taxon_tree"=>{"any"=>["c58fdadd-7743-46d6-9629-90bb3ccc4ef0"]}},
  email_document_supertype: "",
  government_document_supertype: "",
  signon_user_uid: nil,
  slug: "education-training-and-skills",
  url: "/education",
  tags_digest: nil,
  links_digest: "1c115d268a3d4a4dfb7875470573d76793ddc645cadbb3cd3ed34188da2d4237",
  content_id: nil,
  description: nil,
  last_audited_at: Wed, 24 Apr 2024 13:31:57.000000000 BST +01:00>
```
See on line 18 below

**Review by hannako — COMMENTED** [review]
Thanks for reworking the specs Mat! Sorry one more thing. Have only commented in a couple of places, but probably applies in most cases where you've used `taxon` instead of `taxon_tree`

**Comment by unoduetre** [comment]
@hannako 

I've checked and `ContentChange` also uses `taxon_tree`. This is added by [this code](https://github.com/alphagov/email-alert-service/blob/c2bbc08a16af48cd4b5f4fd8dcc57ca533fdbee5/email_alert_service/models/document_links.rb#L13) in `email-alert-service`. This simplified things a bit. If a test already uses `taxon_tree`, I've used `document_collections` instead, as those are also used by the database entries.

File: spec/features/weekly_digest_spec.rb (hunk 2 above)
Comment (unoduetre) [reply spec/features/weekly_digest_spec.rb]: Fixed

File: spec/integration/browsing_subscriber_lists_spec.rb (hunk 3 above)
Comment (unoduetre) [reply spec/integration/browsing_subscriber_lists_spec.rb]: Fixed

File: spec/features/daily_digest_spec.rb (hunk 4 above)
Comment (unoduetre) [reply spec/features/daily_digest_spec.rb]: Fixed

File: spec/features/daily_digest_spec.rb [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
-  let(:list_one_topic_id) { "0eb5d0f0-d384-4f27-9da8-3f9e9b22a820" }
+  let(:list_one_document_collection_id) { "0eb5d0f0-d384-4f27-9da8-3f9e9b22a820" }
   let(:list_two_taxon_id) { "6416e4e0-c0c1-457a-8337-4bf8ed9d5f80" }
 
   let(:subscriber_list_one) do
     create_subscriber_list(
       title: "Subscriber list one",
       links: {
-        topics: { any: [list_one_topic_id] },
+        document_collections: { any: [list_one_document_collection_id] },
```
Comment (hannako) [inline spec/features/daily_digest_spec.rb:9]: Oh no, sorry Mat. You're right in that `document_collections` is a valid `links` key, but a subscriber list would only have this key if it also had a content id. It's not possible to have a link to a document collection without a content id. There are docs [here](https://github.com/alphagov/email-alert-api/blob/main/docs/matching-content-to-subscriber-lists.md#uuid-field) which I added last year - that I now realise are really not clear. So I will take an action to improve those! How about you pick `organisations` instead?

Comment (unoduetre) [reply spec/features/daily_digest_spec.rb]: Fixed. I've used organisations in most places. In a few o them I've used people or countries, as discussed.

**Review by hannako — APPROVED** [review]
Thank Mat, and nice work on removing references to policies as well.

File: spec/lib/reports/subscriber_lists_report_spec.rb [hunk 6]
```diff
@@ -15,7 +15,7 @@
   end
 
   it "returns data around active lists for the given date" do
-    expected_criteria_bits = '{"document_type":"","tags":{"topics":{"any":["motoring/road_rage"]}},' \
+    expected_criteria_bits = '{"document_type":"","tags":{"tribunal_decision_categories":{"any":["motoring/road_rage"]}},' \
```
[the remark this answers is not in the document]
Comment (unoduetre) [reply spec/lib/reports/subscriber_lists_report_spec.rb:18]: I'll check the db data and use some more realistic strings.

[the remark this answers is not in the document]
Comment (unoduetre) [reply spec/lib/reports/subscriber_lists_report_spec.rb]: I've used more realistic data in the tests.

[the remark this answers is not in the document]
Comment (KludgeKML) [reply spec/lib/reports/subscriber_lists_report_spec.rb]: Thanks!

**Review by KludgeKML — APPROVED** [review]
LGTM (assuming the pact tests get fixed)

**Pull request merged**